packages/flutter_tools/test/src/mock_devices.dart

```diff
@@ -0,0 +1,96 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'package:flutter_tools/src/build_info.dart';
+import 'package:flutter_tools/src/device.dart';
+import 'package:flutter_tools/src/project.dart';
+
+/// A list of fake devices to test JSON serialization
+/// (`Device.toJson()` and `--machine` flag for `devices` command)
+List<MockDeviceJson> mockDevices = <MockDeviceJson>[
+  MockDeviceJson(
+    MockDevice('ephemeral', 'ephemeral', true),
+    <String, Object>{
+      'name': 'ephemeral',
+      'id': 'ephemeral',
+      'isSupported': true,
+      'targetPlatform': 'android-arm',
+      'emulator': true,
+      'sdk': 'Test SDK (1.2.3)',
+      'capabilities': <String, Object>{
+        'hotReload': true,
+        'hotRestart': true,
+        'screenshot': false,
+        'fastStart': false,
+        'flutterExit': true,
+        'hwRendering': true,
+        'startPaused': true
+      }
+    }
+  ),
+  MockDeviceJson(
+    MockDevice('webby', 'webby')
+      ..targetPlatform = Future<TargetPlatform>.value(TargetPlatform.web_javascript)
+      ..sdkNameAndVersion = Future<String>.value('Web SDK (1.2.4)'),
+    <String,Object>{
+      'name': 'webby',
+      'id': 'webby',
+      'isSupported': true,
+      'targetPlatform': 'web-javascript',
+      'emulator': true,
+      'sdk': 'Web SDK (1.2.4)',
+      'capabilities': <String, Object>{
+        'hotReload': true,
+        'hotRestart': true,
+        'screenshot': false,
+        'fastStart': false,
+        'flutterExit': true,
+        'hwRendering': false,
+        'startPaused': true
+      }
+    }
+  ),
+];
+
+/// Fake device to test `devices` command
+class MockDevice extends Device {
```

Comment:
Rename this to `FakeDevice` since it doesn't implement the Mock interface anymore